c++/src/kj/common.h

```diff
@@ -2141,6 +2141,12 @@ constexpr bool isDisallowedInCoroutine() {
 // }
 //
 
+template<typename T>
+inline void zero(T& t) {
```

Comment:
Should we limit this so you can only call it on types with trivial constructors and destructors? Otherwise it is ironically unsafe, but not as obviously so as memset. E.g. I could imagine someone trying to zero a `kj::Array`, thinking that it'll zero the contents of the array.

Could this be, like, `memzero` instead of `zero`? Might be clearer what it does that way.

Reply:
added `KJ_HAS_TRIVIAL_CONSTRUCTOR`. Also it is easy to add `kj:zero` override for Array which I did. Should I remove it and rename to `memzero`?

Reply:
renamed to memzero and removed Array version. PTAL.